The job **`x86_64-gnu-nopt`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/9343292758/job/25712665945) [(plain)](https://github.com/rust-lang-ci/rust/commit/f7600557bb8c29eb6bf8634e265e6a9e65833e5a/checks/25712665945/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [codegen] tests/codegen/simd/packed-simd.rs#noopt stdout ----

error in revision `noopt`: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/simd/packed-simd.noopt/packed-simd.ll" "/checkout/tests/codegen/simd/packed-simd.rs" "--check-prefix=CHECK" "--check-prefix" "noopt" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
--- stderr -------------------------------
/checkout/tests/codegen/simd/packed-simd.rs:40:16: error: noopt-DAG: expected string not found in input
 // noopt-DAG: alloca{{.*}} align 4
               ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/simd/packed-simd.noopt/packed-simd.ll:114:6: note: scanning from here
     ^
     ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/simd/packed-simd.noopt/packed-simd.ll:115:16: note: possible intended match here
 %0 = alloca [16 x i8], align 16

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/simd/packed-simd.noopt/packed-simd.ll
Check file: /checkout/tests/codegen/simd/packed-simd.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
          .
          .
          .
          .
         14:  
         15: bb3: ; preds = %start 
         16:  %0 = mul i64 %count, 12 
         17:  call void @llvm.memcpy.p0.p0.i64(ptr align 4 %dst, ptr align 4 %src, i64 %0, i1 false) 
         18:  ret void 
         19:  
         20: bb1: ; No predecessors! 
         21:  unreachable 
         22:  
         23: bb2: ; No predecessors! 
         24:  unreachable 
         25: } 
         27: ; core::mem::maybe_uninit::MaybeUninit<T>::as_mut_ptr 
         27: ; core::mem::maybe_uninit::MaybeUninit<T>::as_mut_ptr 
         28: ; Function Attrs: alwaysinline nonlazybind uwtable 
         29: define internal ptr @"_ZN4core3mem12maybe_uninit20MaybeUninit$LT$T$GT$10as_mut_ptr17h00ee2515e4605f05E"(ptr align 16 %self) unnamed_addr #0 { 
         30: start: 
         31:  ret ptr %self 
         32: } 
         34: ; core::mem::maybe_uninit::MaybeUninit<T>::assume_init 
         34: ; core::mem::maybe_uninit::MaybeUninit<T>::assume_init 
         35: ; Function Attrs: alwaysinline nonlazybind uwtable 
         36: define internal void @"_ZN4core3mem12maybe_uninit20MaybeUninit$LT$T$GT$11assume_init17hd0b0ffc366cbe5f2E"(ptr sret([16 x i8]) align 16 %_0, ptr align 16 %self, ptr align 8 %0) unnamed_addr #0 { 
         37: start: 
         38:  %slot = load <3 x float>, ptr %self, align 16 
         39:  store <3 x float> %slot, ptr %_0, align 16 
         40:  ret void 
         41: } 
         43: ; core::mem::maybe_uninit::MaybeUninit<T>::uninit 
         43: ; core::mem::maybe_uninit::MaybeUninit<T>::uninit 
         44: ; Function Attrs: alwaysinline nonlazybind uwtable 
         45: define internal void @"_ZN4core3mem12maybe_uninit20MaybeUninit$LT$T$GT$6uninit17h49faceeb97a9d509E"(ptr sret([16 x i8]) align 16 %_0) unnamed_addr #0 { 
         47:  ret void 
         48: } 
         49:  
         50: ; core::ptr::mut_ptr::<impl *mut T>::cast 
         50: ; core::ptr::mut_ptr::<impl *mut T>::cast 
         51: ; Function Attrs: alwaysinline nonlazybind uwtable 
         52: define internal ptr @"_ZN4core3ptr7mut_ptr31_$LT$impl$u20$$BP$mut$u20$T$GT$4cast17hedbd2cf60969663dE"(ptr %self) unnamed_addr #0 { 
         53: start: 
         54:  ret ptr %self 
         55: } 
         57: ; packed_simd::load 
         57: ; packed_simd::load 
         58: ; Function Attrs: nonlazybind uwtable 
         59: define internal void @_ZN11packed_simd4load17h2d1fda06ec35af9eE(ptr sret([16 x i8]) align 16 %_0, ptr align 4 %v) unnamed_addr #1 personality ptr @rust_eh_personality { 
         60: start: 
         61:  %0 = alloca [16 x i8], align 16 
         62:  %1 = alloca [16 x i8], align 8 
         63:  %tmp = alloca [16 x i8], align 16 
         64: ; invoke core::mem::maybe_uninit::MaybeUninit<T>::uninit 
         65:  invoke void @"_ZN4core3mem12maybe_uninit20MaybeUninit$LT$T$GT$6uninit17h49faceeb97a9d509E"(ptr sret([16 x i8]) align 16 %tmp) 
         66:  to label %bb1 unwind label %cleanup 
         67:  
         68: bb7: ; preds = %cleanup 
         69:  %2 = load ptr, ptr %1, align 8 
         70:  %3 = getelementptr inbounds i8, ptr %1, i64 8 
         71:  %4 = load i32, ptr %3, align 8 
         72:  %5 = insertvalue { ptr, i32 } poison, ptr %2, 0 
         73:  %6 = insertvalue { ptr, i32 } %5, i32 %4, 1 
         74:  resume { ptr, i32 } %6 
         75:  
         76: cleanup: ; preds = %bb4, %bb3, %bb2, %bb1, %start 
         77:  %7 = landingpad { ptr, i32 } 
         78:  cleanup 
         79:  %8 = extractvalue { ptr, i32 } %7, 0 
         80:  %9 = extractvalue { ptr, i32 } %7, 1 
         81:  store ptr %8, ptr %1, align 8 
         82:  %10 = getelementptr inbounds i8, ptr %1, i64 8 
         83:  store i32 %9, ptr %10, align 8 
         84:  br label %bb7 
         85:  
         86: bb1: ; preds = %start 
         87: ; invoke core::mem::maybe_uninit::MaybeUninit<T>::as_mut_ptr 
         88:  %_7 = invoke ptr @"_ZN4core3mem12maybe_uninit20MaybeUninit$LT$T$GT$10as_mut_ptr17h00ee2515e4605f05E"(ptr align 16 %tmp) 
         89:  to label %bb2 unwind label %cleanup 
         90:  
         91: bb2: ; preds = %bb1 
         92: ; invoke core::ptr::mut_ptr::<impl *mut T>::cast 
         93:  %_6 = invoke ptr @"_ZN4core3ptr7mut_ptr31_$LT$impl$u20$$BP$mut$u20$T$GT$4cast17hedbd2cf60969663dE"(ptr %_7) 
         94:  to label %bb3 unwind label %cleanup 
         95:  
         96: bb3: ; preds = %bb2 
         97: ; invoke core::intrinsics::copy_nonoverlapping 
         98:  invoke void @_ZN4core10intrinsics19copy_nonoverlapping17h92fe1b597bfddadaE(ptr %v, ptr %_6, i64 1) 
         99:  to label %bb4 unwind label %cleanup 
        100:  
        101: bb4: ; preds = %bb3 
        102:  %_9 = load <3 x float>, ptr %tmp, align 16 
        103:  store <3 x float> %_9, ptr %0, align 16 
        104: ; invoke core::mem::maybe_uninit::MaybeUninit<T>::assume_init 
        105:  invoke void @"_ZN4core3mem12maybe_uninit20MaybeUninit$LT$T$GT$11assume_init17hd0b0ffc366cbe5f2E"(ptr sret([16 x i8]) align 16 %_0, ptr align 16 %0, ptr align 8 @alloc_e87509d2688b3ffec082f76e8b47608f) 
        106:  to label %bb5 unwind label %cleanup 
        107:  
        108: bb5: ; preds = %bb4 
        109:  ret void 
        110: } 
        111:  
        112: ; Function Attrs: nonlazybind uwtable 
        113: define void @square_packed(ptr sret([16 x i8]) align 16 %_0, ptr align 4 %x) unnamed_addr #1 { 
        114: start: 
dag:40'0          X~ error: no match found
        115:  %0 = alloca [16 x i8], align 16 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dag:40'1                    ?                  possible intended match
        116: ; call packed_simd::load 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
        117:  call void @_ZN11packed_simd4load17h2d1fda06ec35af9eE(ptr sret([16 x i8]) align 16 %0, ptr align 4 %x) 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        118:  %x1 = load <3 x float>, ptr %0, align 16 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        119:  %1 = fmul <3 x float> %x1, %x1 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        120:  store <3 x float> %1, ptr %_0, align 16 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        121:  ret void 
dag:40'0     ~~~~~~~~~~
        122: } 
dag:40'0     ~~
dag:40'0     ~
dag:40'0     ~
        124: ; Function Attrs: nocallback nofree nounwind willreturn memory(argmem: readwrite) 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        125: declare void @llvm.memcpy.p0.p0.i64(ptr noalias nocapture writeonly, ptr noalias nocapture readonly, i64, i1 immarg) #2 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dag:40'0     ~
        127: ; Function Attrs: nonlazybind 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        128: declare i32 @rust_eh_personality(...) unnamed_addr #3 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dag:40'0     ~
dag:40'0     ~
        130: attributes #0 = { alwaysinline nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        131: attributes #1 = { nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        132: attributes #2 = { nocallback nofree nounwind willreturn memory(argmem: readwrite) } 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        133: attributes #3 = { nonlazybind "target-cpu"="x86-64" } 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dag:40'0     ~
dag:40'0     ~
        135: !llvm.module.flags = !{!0, !1} 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        136: !llvm.ident = !{!2} 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~
dag:40'0     ~
dag:40'0     ~
        138: !0 = !{i32 8, !"PIC Level", i32 2} 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        139: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        140: !2 = !{!"rustc version 1.80.0-nightly (f7600557b 2024-06-03)"} 
dag:40'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

--- a/tests/codegen/simd/packed-simd.rs
+++ b/tests/codegen/simd/packed-simd.rs
@@ -1,0 +1,44 @@
+//@ revisions:opt3 noopt
+//@[opt3] compile-flags: -Copt-level=3
+//@[noopt] compile-flags: -Cno-prepopulate-passes
+
+#![crate_type = "lib"]
+#![no_std]
+#![feature(repr_simd, core_intrinsics)]
+use core::intrinsics::simd as intrinsics;
+use core::{mem, ptr};
+
+// Test codegen for not only "packed" but also "fully aligned" SIMD types, and conversion between
+// A repr(packed,simd) type with 3 elements can't exceed its element alignment,
+// whereas the same type as repr(simd) will instead have padding.
+
+#[repr(simd, packed)]
+pub struct Simd<T, const N: usize>([T; N]);
+
+#[repr(simd)]
+#[derive(Copy, Clone)]
+pub struct FullSimd<T, const N: usize>([T; N]);
+
+// non-powers-of-two have padding and need to be expanded to full vectors
+fn load<T, const N: usize>(v: Simd<T, N>) -> FullSimd<T, N> {
+    unsafe {
+        let mut tmp = mem::MaybeUninit::<FullSimd<T, N>>::uninit();
+        ptr::copy_nonoverlapping(&v as *const _, tmp.as_mut_ptr().cast(), 1);
+        tmp.assume_init()
+    }
+}
+
+// CHECK-LABEL: square_packed
+// CHECK-SAME: ptr{{[a-z_ ]*}} sret([[RET_TYPE:[^)]+]]) [[RET_ALIGN:align (8|16)]]{{[^%]*}} [[RET_VREG:%[_0-9]*]]
+// CHECK-SAME: ptr{{[a-z_ ]*}} align 4
+#[no_mangle]
+pub fn square_packed(x: Simd<f32, 3>) -> FullSimd<f32, 3> {
+    // CHECK-NEXT: start
+    // noopt: alloca [[RET_TYPE]], [[RET_ALIGN]]
+    // CHECK: load <3 x float>
+    let x = load(x);
+    // CHECK: [[VREG:%[a-z0-9_]+]] = fmul <3 x float>
+    // CHECK-NEXT: store <3 x float> [[VREG]], ptr [[RET_VREG]], [[RET_ALIGN]]
+    // CHECK-NEXT: ret void
+    unsafe { intrinsics::simd_mul(x, x) }
+}

